compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -4041,6 +4050,58 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn detect_and_explain_multiple_crate_versions(
+        &self,
+        err: &mut Diag<'_>,
+        item: &ty::AssocItem,
+        owner: hir::OwnerId,
+        rcvr_ty: Ty<'_>,
+    ) -> bool {
+        let pick_name = self.tcx.crate_name(item.def_id.krate);
+        if let Some(map) = self.tcx.in_scope_traits_map(owner) {
+            for trait_candidate in map.to_sorted_stable_ord().into_iter().flat_map(|v| v.1.iter()) {
+                let name = self.tcx.crate_name(trait_candidate.def_id.krate);
+                if trait_candidate.def_id.krate != item.def_id.krate && name == pick_name {
+                    let msg = format!(
+                        "you have multiple different versions of crate `{name}` in your \
```

Comment:
I feel like saying "you" here makes this read weirdly

```suggestion
                        "there have multiple different versions of crate `{name}` in the \
```

Reply:
There are a couple of other strings in this module that use `you`, we should clean those up too, separately.